File: 006_HAMM/006_HAMM.py
'''
Title: Counting Point Mutations
Rosalind ID: HAMM
Rosalind #: 006
URL: http://rosalind.info/problems/hamm
'''

f = open("./rosalind_hamm.txt", 'r')
raw_data = f.readlines()
f.close()

# Eliminar saltos de línea y espacios en blanco
seq1 = raw_data[0].strip()
seq2 = raw_data[1].strip()

# Verificar que las secuencias tengan la misma longitud
if len(seq1) != len(seq2):
    print("Error: Las secuencias tienen longitudes diferentes.")
    exit(1)

mutations = 0  # Inicializar contador de mutaciones

# Iterar a través de las secuencias y contar las diferencias
for i in range(len(seq1)):
    if seq1[i] != seq2[i]:
        mutations += 1

print(mutations)

File: 006_HAMM/006_HAMM.c

/*
Title: Counting Point Mutations
Rosalind ID: HAMM
Rosalind #: 006
URL: http://rosalind.info/problems/hamm
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_LINE_LENGTH 1000

int main() {
    FILE *file = fopen("rosalind_hamm.txt", "r");
    if (file == NULL) {
        perror("Error opening file");
        return 1;
    }

    char seq1[MAX_LINE_LENGTH];
    char seq2[MAX_LINE_LENGTH];

    // Leer las dos secuencias
    fgets(seq1, MAX_LINE_LENGTH, file);
    fgets(seq2, MAX_LINE_LENGTH, file);

    fclose(file);

    // Eliminar saltos de línea
    seq1[strcspn(seq1, "\n")] = '\0';
    seq2[strcspn(seq2, "\n")] = '\0';

    // Verificar que las secuencias tengan la misma longitud
    if (strlen(seq1) != strlen(seq2)) {
        printf("Error: Las secuencias tienen longitudes diferentes.\n");
        return 1;
    }

    int mutations = 0;

    // Contar las diferencias
    for (int i = 0; i < strlen(seq1); i++) {
        if (seq1[i] != seq2[i]) {
            mutations++;
        }
    }

    printf("%d\n", mutations);

    return 0;
}